UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformationAdmin.Web/Controllers/RadioNavigationalWarningsAdminController.cs

```diff
@@ -39,16 +39,30 @@ public async Task<IActionResult> Create(RadioNavigationalWarning radioNavigation
         {
             _logger.LogInformation(EventIds.CreateNewRNWRecordStart.ToEventId(), "Create RNW request started for _X-Correlation-ID:{correlationId}. Requested by user: {user}", GetCurrentCorrelationId(),User.Identity.Name);
 
+            bool skipCheckDuplicateReference = false;
+            if (!string.IsNullOrWhiteSpace(Request.Form["SkipCheckDuplicate"]))
+            {
+                skipCheckDuplicateReference = Request.Form["SkipCheckDuplicate"] == "Yes";
+            }
+
             if (ModelState.IsValid)
             {
-                bool result = await _rnwService.CreateNewRadioNavigationWarningsRecord(radioNavigationalWarning, GetCurrentCorrelationId());
+                ResponseNewRadioNavigationWarningsModel result = await _rnwService.CreateNewRadioNavigationWarningsRecord(radioNavigationalWarning, GetCurrentCorrelationId(), skipCheckDuplicateReference, User.Identity.Name);
 
-                if (result)
+                if (result.IsCreated)
```

Comment:
The name of this property is a little misleading with respect to how it is currently used.

While it is true that it will not have created for a duplicate, it is only true as a result of a check for a duplicate and is only used to prevent duplication, where the 'false' path is an option to clone or cancel (preventing a duplicate).

A better name might be HasDuplicate, or possibly WillHaveDuplicate (as the duplication is yet to occur)?

Reply:
The reason I stress this is that for a moment I questioned whether creation is a suitable duplicate check - a more thorough review proved otherwise but as the code is maintained going forward this could be further obscured and make maintenance more difficult.

Reply:
Thanks for highlighting done required changes